tests/test_cli.py

```diff
@@ -182,3 +191,169 @@ def test_find_designated_time_function(self):
+            [call(dst='/dst_folder/', src='/testing_folder/testing_file.test'),
+             call(dst='/dst_folder/', src='/new_loc_add')],
+            mock_copy.call_args_list)
+        for dataset in Dataset.objects.all():
+            dataset.dataseturi_set.filter(
+                uri='/new_loc_add', dataset=dataset, service=LOCAL_FILE_SERVICE).delete()
+
+    @mock.patch('geospaas_processing.cli.copy.path')
+    def test_correct_place_of_symlink_after_creation_of_it(self, mock_isfile):
+        """ symlink must be placed at the address that is specified from the input arguments """
+        mock_isfile.isfile.return_value = True
+        sys.argv = [
+            "",
+            '-b', "2018-04-01",
+            '-e', "2018-04-09",
+            '-l',
+            '-d', "/dst_folder/"
+        ]
+        for dataset in Dataset.objects.all():
+            dataset.dataseturi_set.get_or_create(
+                uri='/new_loc_add', dataset=dataset, service=LOCAL_FILE_SERVICE)
+        with mock.patch('os.symlink') as mock_symlink:
+            cli_copy.main()
+        self.assertEqual(
+            [call(dst='/dst_folder/testing_file.test', src='/testing_folder/testing_file.test'),
+             call(dst='/dst_folder/new_loc_add', src='/new_loc_add')],
+            mock_symlink.call_args_list)
+        for dataset in Dataset.objects.all():
+            dataset.dataseturi_set.filter(
+                uri='/new_loc_add', dataset=dataset, service=LOCAL_FILE_SERVICE).delete()
+
+    @mock.patch('os.symlink')
+    @mock.patch('geospaas_processing.cli.copy.path')
+    def test_correct_content_of_flag_file(self, mock_isfile, mock_link):
+        """ flag file should contain this 'type: test_type' information """
+        mock_isfile.isfile.return_value = True
+        sys.argv = [
+            "",
+            '-b', "2018-04-01",
+            '-e', "2018-04-09",
+            '-f', '-l',
+            '-t', 'test_type'
+        ]
+        temp_directory = tempfile.TemporaryDirectory()
+        sys.argv.append('-d')
+        sys.argv.append(temp_directory.name)
+        cli_copy.main()
+        fd = os.open(temp_directory.name + '/testing_file.test.flag', os.O_RDONLY)
+        self.assertEqual(os.read(fd, 200),
+                         b"type: test_type\nurl: https://scihub.copernicus.eu/apihub/odata/v1/Produ"
+                         + b"cts('6127111d-c9bd-4689-bab5-412dd39e1e81')/$value\nurl: https://scihu"
+                         + b"b.copernicus.eu/the_second_fakeurl\n")
+
+    def test_copying_accomplishment_and_consistency(self):
+        """ A symlink of actual sample file should be copied to the destination folder """
+        sys.argv = [
+            "",
+            '-b', "2018-04-01",
+            '-e', "2018-04-09",
+            '-f', '-l'
+        ]
+        temp_directory_dst = tempfile.TemporaryDirectory()
+        sys.argv.append('-d')
+        sys.argv.append(temp_directory_dst.name)
+        os.mkdir("/testing_folder")
```

Comment:
Don't create files and directories under `/`.
You created a temporary directory, use it: you can have both your source and destination folders in it.

Reply:
The fixture are installed in the database. The address in the fixture is constant and we can change it based various execution of testing suite. So it is necessary to create a real file in a real folder for testing purposes!

We can change the destination folder based on the tempfolder at the time of execution of the tests. But this is not doable for the source address because source address comes from the database.

Reply:
Then at least create it in `/tmp`.

Reply:
Done in the next commit

Reply:
Done in the next commit.